micro-rdk/src/esp32/entry.rs

```diff
@@ -164,7 +153,16 @@ pub fn serve_web(
     let exec = Esp32Executor::new();
     let cloned_exec = exec.clone();
 
-    let fut = cloned_exec.run(Box::pin(serve_web_inner(
+    cloned_exec
+        .spawn(async {
+            loop {
+                Timer::after(Duration::from_secs(150)).await;
+                unsafe { crate::esp32::esp_idf_svc::sys::esp_task_wdt_reset() };
+            }
+        })
+        .detach();
+
+    cloned_exec.run_forever(Box::pin(serve_web_inner(
```

Comment:
Why is the `Box::pin` required?

Reply:
It will force the state machine to be allocated on the ram vs stack potentially